crates/bevy_window/src/event.rs

```diff
@@ -44,19 +38,16 @@ pub struct WindowCreated {
 ///
 /// [`WindowPlugin`]: crate::WindowPlugin
 /// [`Window`]: crate::Window
-/// [closing]: crate::Window::close
 #[derive(Debug, Clone)]
 pub struct WindowCloseRequested {
-    pub id: WindowId,
+    pub entity: Entity,
 }
 
 /// An event that is sent whenever a window is closed. This will be sent by the
-/// handler for [`Window::close`].
-///
-/// [`Window::close`]: crate::Window::close
+/// handler for [`WindowCloseRequested`] or similar.
 #[derive(Debug, Clone)]
 pub struct WindowClosed {
-    pub id: WindowId,
+    pub entity: Entity,
 }
```

Comment:
We should rethink this event's purpose or rename it. 

Right now it is being used as "Hey winit, please close & despawn this window". If that is the desired behaviour it should be renamed to `DespawnWindow` / `DestroyWindow` or something of that nature

I think we _also_ should have an event which captures "Winit has now destroyed / despawned the window". 
_That_ event should be named `WindowClosed` / `WindowDestroyed` since it captures an event that _has_ happened

Reply:
This should be the case now, despawning a window entity will close it rather than having to send a specific event.